tools/applicationinsights-web-snippet/src/snippet.ts

```diff
@@ -202,11 +204,60 @@ declare var cfg:ISnippetConfig;
             "js2.cdn.applicationinsights.io",
             "js2.cdn.monitor.azure.com",
             "az416426.vo.msecnd.net" // this domain is supported but not recommended
-        ]
-    
-        // Assigning these to local variables allows them to be minified to save space:
-        let targetSrc : string = (aiConfig as any)["url"] || cfg.src
-        if (targetSrc) {
+        ];
+
+        let targetSrc : string = (aiConfig as any)["url"] || cfg.src;
+        var match = targetSrc.match(/^(http[s]?:\/\/.*\/)([\w]+\.)?(\d+(\.\d+){0,2})\.(([\w]+\.){0,2}js)$/);
+        if (match.length === 7 && cfg.sri) {
+            var integrityUrl = match[1] + "ai." + match[3] + ".integrity.json";
```

Comment:
We don't really want to depend on the "name" being `ai.` here either...

Reply:
if you add an extra group around your regex you can get another group match which includes everything except the the end bit like `.min.js` and then use that to just append the `.integrity.json`

((http[s]?:\/\/.*\/)([\w]+\.)?(\d+(\.\d+){0,2}))\.(([\w]+\.){0,2}js)$